### PR TITLE
Remove `http-stream-request-body`

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
This is another spin-off from #4903 and it removes `http-stream-request-body`. Two reasons:

1. As we move towards support for streaming throughout the runtime, that flag will not be needed anymore
2. The flag doesn't work today anyways. Per my research on the internals of fasthttp (from #4979):
   1. Setting that flag enables consumers to read data as stream (with the caveats from #4979) but currently Dapr doesn't use the methods to read data as streams: Dapr only uses methods like `reqCtx.PostBody()` which return the entire request body as a byte slice, regardless of streaming support.
   2. That flag causes some unintended behavior, such as `http-max-request-size` to not work as expected

Docs issue: https://github.com/dapr/docs/issues/2737

### CODE REVIEW

File: pkg/messaging/direct_messaging.go [hunk 1]
```diff
@@ -61,7 +60,6 @@ type directMessaging struct {
 	grpcPort            int
 	namespace           string
 	resolver            nr.Resolver
-	tracingSpec         config.TracingSpec
```
Comment (ItalyPaleAle) [inline pkg/messaging/direct_messaging.go:64]: Removed as it was not used

File: pkg/messaging/direct_messaging.go [hunk 2]
```diff
@@ -77,45 +75,47 @@ type remoteApp struct {
...
-	resolver nr.Resolver,
-	tracingSpec config.TracingSpec,
-	maxRequestBodySize int,
-	proxy Proxy,
-	readBufferSize int,
-	streamRequestBody bool,
-	resiliency resiliency.Provider,
-	isResiliencyEnabled bool,
-) DirectMessaging {
+func NewDirectMessaging(opts NewDirectMessagingOpts) DirectMessaging {
```
Comment (ItalyPaleAle) [inline pkg/messaging/direct_messaging.go:95]: Why this change: the old method's signature had 15 arguments and it was very hard to maintain when we needed to remove or add parameters (like in this case), especially in tests

**Comment by ItalyPaleAle** [comment]
Now that the other PR is merged, this is ready for review

File: pkg/messaging/direct_messaging.go (hunk 2 above)
Comment (berndverst) [reply pkg/messaging/direct_messaging.go:95]: There are some slight performance difference in using the options pattern vs the named params. But I don't recall which one is better haha

**Review by berndverst — APPROVED** [review]

**Comment by berndverst** [comment]
LGTM

File: pkg/messaging/direct_messaging.go (hunk 2 above)
Comment (ItalyPaleAle) [reply pkg/messaging/direct_messaging.go:95]: This may be slightly worse, but this code path is only executed once during initialization.

**Review by yaron2 — APPROVED** [review]
